### PR TITLE
Use new currency encoding in configs and genesis ledger

### PR DESCRIPTION
**Pull request by nholland94**

### CODE REVIEW

**Comment by bkase** [comment]
https://app.circleci.com/jobs/github/CodaProtocol/coda/339114 <-- this link has the stack traces and failing tests for this PR

**Comment by bkase** [comment]
We should probably migrate `Currency.of_int` to `Currency.of_string` everywhere in the tests

**Comment by psteckler** [comment]
The `Option.value_exn` exceptions in the staged ledger unit tests when run with `test_postake_snarkless` are due to a `None` in the subtraction in `Quickcheck_lib.gen_division_generic`.

File: src/lib/staged_ledger/staged_ledger.ml [hunk 1]
```diff
@@ -1949,16 +1949,7 @@ let%test_module "test" =
           [%sexp_of:
             Ledger.init_state
             * User_command.With_valid_signature.t list
-            * int option list]
-        ~shrinker:
```
Comment (emberian) [inline src/lib/staged_ledger/staged_ledger.ml]: I'm assuming this shrinker was causing problems?

Comment (psteckler) [reply src/lib/staged_ledger/staged_ledger.ml]: Yes, the test data the shrinker generated triggered an assertion failure in `iter_cmds_acc`.

I'm not sure a shrinker is the right approach for these tests, in any case. If I understand, a shrinker is invoked when there's a failure with the original test data, so invoking the shrinker is masking something with the original test data.

**Comment by yourbuddyconner** [comment]
This PR's got a failing test: 

```
!!! (monitor.ml.Error
!!!  ("Hashtbl.add_exn got key already present"
!!!   ((payload
!!!     ((common
!!!       ((fee 2000000000) (nonce 0) (valid_until 4294967295)
!!!        (memo
!!!         "\000 i!z0y\144\128\148\225\017!\208B5J|\031U\182H,\161\165\030\027%\r\253\030\208\238\249")))
!!!      (body
!!!       (Payment
!!!        ((receiver
!!!          ZsMSUqpv8LP1X888jpyJxvUgUdBUxGB93MvzQ94bwi3FXB2WVLRsE9zG2ZbEGQSu1np)
!!!         (amount 10))))))
!!!    (sender
!!!     ZsMSUvEVbtHdQEQ2f2EY6yp4qgAxBxLmtUx63sFJMdwMn44fAGbMTkeqGqNU2Ms93Me)
!!!    (signature
!!!     (194233195464185559309603147245264067629186332345669911792016498968485887129140338182410865
!!!      137568278676227302130198161054308060330438655016055680581351168922148284533801912411784434))))
!!!  ("Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30"
!!!   "Called from file \"src/hashtbl.ml\", line 140, characters 4-21"
!!!   "Called from file \"src/app/cli/src/tests/coda_worker_testnet.ml\", line 568, characters 24-308"
!!!   "Called from file \"src/deferred1.ml\", line 17, characters 40-45"
!!!   "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
!!!   "Called from file \"src/job_queue.ml\", line 171, characters 6-47"))
ERROR: Test "test_postake_txns:coda-shared-state-test" failed
```

**Comment by psteckler** [comment]
> This PR's got a failing test

Apparently caused by duplicate sample keypairs, so that we have duplicate user commands. 

In `Coda_worker_testnet`, I changed a `Hashtbl.add_exn` to `ignore (Hashtbl.add ...)`. That will fix this particular error, and I don't think it will break anything else.

**Pull request merged**